rig-neo4j/src/lib.rs

```diff
@@ -162,19 +176,165 @@ impl Neo4jClient {
         Ok(Self { graph })
     }
 
+    pub async fn execute_and_collect<T: for<'a> Deserialize<'a>>(
+        graph: &Graph,
+        query: Query,
+    ) -> Result<Vec<T>, VectorStoreError> {
+        graph
+            .execute(query)
+            .await
+            .map_err(neo4j_to_rig_error)?
+            .into_stream_as::<T>()
+            .try_collect::<Vec<T>>()
+            .await
+            .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))
+    }
+
     /// Returns a `Neo4jVectorIndex` that mirrors an existing Neo4j Vector Index.
     ///
     /// An index (of type "vector") of the same name as `index_name` must already exist for the Neo4j database.
     /// See the Neo4j [documentation (Create vector index)](https://neo4j.com/docs/genai/tutorials/embeddings-vector-indexes/setup/vector-index/) for more information on creating indexes.
     ///
     /// ❗IMPORTANT: The index must be created with the same embedding model that will be used to query the index.
-    pub fn index<M: EmbeddingModel>(
+    pub async fn get_index<M: EmbeddingModel>(
         &self,
         model: M,
-        index_config: IndexConfig,
+        index_name: &str,
         search_params: SearchParams,
-    ) -> Neo4jVectorIndex<M> {
-        Neo4jVectorIndex::new(self.graph.clone(), model, index_config, search_params)
+    ) -> Result<Neo4jVectorIndex<M>, VectorStoreError> {
+        #[derive(Deserialize)]
+        struct IndexInfo {
+            name: String,
+            properties: Vec<String>,
+            options: IndexOptions,
+        }
+
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct IndexOptions {
+            _index_provider: String,
+            index_config: IndexConfigDetails,
+        }
+
+        #[derive(Deserialize)]
+        struct IndexConfigDetails {
+            #[serde(rename = "vector.dimensions")]
+            vector_dimensions: i64,
+            #[serde(rename = "vector.similarity_function")]
+            vector_similarity_function: String,
+        }
```

Comment:
Why define these structs within the method here? Doesn't this make exterior usage of these awkward?

Reply:
These structures would be of no use (for now). It's to make the deserialization more readable while keeping the module less clutered. Don't know if there is a better way to do this though.

Reply:
@cvauclair